proccontrol/src/process.C

```diff
@@ -242,13 +244,56 @@ bool int_process::attachThreads()
          }
          pthrd_printf("Creating new thread for %d/%d during attach\n", pid, *i);
          thr = int_thread::createThread(this, NULL_THR_ID, *i, false, int_thread::as_needs_attach);
-         found_new_threads = true;
+         found_new_threads = loop_new_threads = true;
       }
-   } while (found_new_threads);
+   } while (loop_new_threads);
 
    return true;
 }
 
+bool int_process::attachThreads()
+{
+   bool found_new_threads = false;
+   return attachThreads(found_new_threads);
+}
+
+// Attach any new threads and synchronize, until there are no new threads
+bool int_process::attachThreadsSync()
+{
+   while (true) {
+      bool found_new_threads = false;
+
+      ProcPool()->condvar()->lock();
+      bool result = attachThreads(found_new_threads);
+      if (found_new_threads)
+         ProcPool()->condvar()->broadcast();
+      ProcPool()->condvar()->unlock();
+
+      if (!result) {
+         pthrd_printf("Failed to attach to threads in %d\n", pid);
+         setLastError(err_internal, "Could not get threads during attach\n");
+         return false;
+      }
+
+      if (!found_new_threads)
+         return true;
+
+      pthrd_printf("Wait again for attach from process %d\n", pid);
+      bool proc_exited = false;
+      result = waitAndHandleForProc(true, this, proc_exited);
+      if (!result) {
+         perr_printf("Internal error calling waitAndHandleForProc on %d\n", getPid());
+         setLastError(err_internal, "Error while calling waitAndHandleForProc for attached threads\n");
```

Comment:
Drop the setLastError here. The lower-level code should be responsible for doing a setLastError that records a more specific message when it first detects the error.